server/lib/routes/otp.js

```diff
@@ -4,88 +4,116 @@ const uniqid = require('uniqid')
       }
 
-      const { secretId, subscriptionId } = tokenData.data()
-
-      const subscription = await db
-        .collection('subscriptions')
-        .doc(subscriptionId)
-        .get()
+      log.info('Request approved, sending back OTP')
+      return reply.send(otp)
+    }
 
-      if (!subscription.exists) {
-        log.error(`Subscription not found - ${subscriptionId}`)
-        return reply.notFound('Subscription not found')
+    const unsubscribe = requestObj.onSnapshot(
+      (update) => {
+        const approved = update.get('approved')
+        if (approved === undefined) {
+          // update due to object creation
+          return
+        }
+        completeRequest(null, update.get('otp'))
+      },
+      (error) => {
+        log.error(error.message)
+        return reply.internalServerError()
       }
+    )
 
-      const uniqueId = uniqid()
+    const { name, version } = body.packageInfo
 
-      return new Promise(() => {
-        const requestObj = db.collection('requests').doc(uniqueId)
-        requestObj.set({ createdAt: new Date() })
+    const notification = {
+      subscription: subscription.data(),
+      secretId,
+      uniqueId,
+      token
+    }
+
+    const hasPackageInfo = !!name || !!version
 
-        const completeRequest = (error, otp) => {
-          requestObj.delete()
-          if (error) {
-            log.error(error.message)
-            return reply.internalServerError()
-          }
+    if (hasPackageInfo) {
+      notification.packageInfo = {
+        ...(version && { version }),
+        ...(name && { name })
+      }
+    }
 
-          unsubscribe()
-          clearTimeout(timeout)
+    push.send(notification)
 
-          if (!otp) {
-            log.error('Request was not approved')
-            return reply.forbidden('Request was not approved')
-          }
+    const timeout = setTimeout(completeRequest, approvalLimit)
+  })
+}
 
-          log.info('Request approved, sending back OTP')
-          return reply.send(otp)
-        }
+async function otpRoutes(server) {
+  server.route({
+    method: 'POST',
+    url: '/api/generate/:token',
+    handler: generateOtpHandler(server)
+  })
 
-        const unsubscribe = requestObj.onSnapshot(
-          (update) => {
-            const approved = update.get('approved')
-            if (approved === undefined) {
-              // update due to object creation
-              return
-            }
-            completeRequest(null, update.get('otp'))
-          },
-          (error) => {
-            log.error(error.message)
-            return reply.internalServerError()
-          }
-        )
-
-        push.send({
-          subscription: subscription.data(),
-          secretId,
-          uniqueId,
-          token
-        })
-
-        const timeout = setTimeout(completeRequest, approvalLimit)
-      })
-    }
+  /**
+   * @deprecated From now on the generate otp route will be a POST request
```

Comment:
is there a way to deprecate routes in fastify?

Reply:
I don't think so, just looked at their docs and couldn't find anything.